lib/storage/client.js

```diff
@@ -1182,6 +1183,267 @@ class Client extends CryptoConsumer {
+        membership_key: member.membershipKey,
+        capability_names: member.capabilities,
+      })
+    })
+    let addGroupMembersRequest = {
+      group_id: groupId,
+      group_members: groupMembersToAdd,
+    }
+    // Call the API endpoint to add group members
+    let response = await this.authenticator.tsv1Fetch(
+      `${this.config.apiUrl}/v2/storage/groups/${groupId}/members`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(addGroupMembersRequest),
+      }
+    )
+    // Await the Response from the API with the Added Group member Response
+    let groupMembersJson = await validateResponseAsJSON(response)
+    let listGroupMembers = []
+    groupMembersJson.forEach(member => {
+      listGroupMembers.push(member)
+    })
+    return listGroupMembers
+  }
+
+  /**
+   * removeGroupMembers removes members to the TozStore Group given.
+   *
+   * @param {string} groupId  UUID assigned by TozStore, used to identify a group.
+   * @param {Array} clientIds all the clientIDs corresponding to the group members you want to remove
+   *
+   *
+   * @returns {Promise<bool>}
+   *
+   */
+  async removeGroupMembers(groupId, clientIds = []) {
+    // Take the Given Paramaters and create the object needed for endpoint
+    let groupMemberToRemove = []
+    clientIds.forEach(client => {
+      groupMemberToRemove.push({
+        client_id: client,
+      })
+    })
+    let removeGroupMembersRequest = {
+      group_id: groupId,
+      group_members: groupMemberToRemove,
+    }
+    // Call the API endpoint to remove group members
+    let response = await this.authenticator.tsv1Fetch(
+      `${this.config.apiUrl}/v2/storage/groups/${groupId}/members`,
+      {
+        method: 'DELETE',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(removeGroupMembersRequest),
+      }
+    )
+    // Check the response code
+    switch (response.status) {
+      case 204:
+        return true
+      case 403:
+        throw new Error('Unauthorized')
+      default:
+        throw new Error('Error while deleting record data.')
+    }
+  }
+  /**
+   * listGroupMembers lists members of the TozStore Group given.
+   *
+   * @param {string} groupId  UUID assigned by TozStore, used to identify a group.
+   *
+   * @returns {GroupMember[]} A response from TozStore; the group members of the group.
+   */
+  async listGroupMembers(groupId) {
+    // Call the API endpoint to remove group members
+    let response = await this.authenticator.tsv1Fetch(
+      `${this.config.apiUrl}/v2/storage/groups/${groupId}/members`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    )
+    let groupMembersJson = await validateResponseAsJSON(response)
+    let listGroupMembers = []
+    groupMembersJson.forEach(member => {
```

Comment:
Is this necessary, aren't you just iterating over a list and pushing contents onto another list?